### ISSUE TITLE
All iteration scores are NaN

### ISSUE THREAD

**Issue by crockpotveggies** [deeplearning4j/deeplearning4j#1462]
![screen shot 2016-04-28 at 11 40 43 am](https://cloud.githubusercontent.com/assets/1445295/14897836/7d66934c-0d38-11e6-9c25-66ef2df15e3f.png)

I'm not able to view the score vs. iteration on the latest master when using either `regularization(true)` and `regularization(false)`. I'm using the `EarlyStoppingTrainer` on a dataset of <10k black and white JPEGs and here are my configs:

Convolutional net: https://gist.github.com/crockpotveggies/d342bbb8a614fc88286bb8e75a8454e1
Training implementation: https://gist.github.com/crockpotveggies/54fd6c86bccc4822ddf2f1c8f0a3d42c

**Comment by crockpotveggies** [comment]
Confirmed it's not related to the UI. Put in a quick `ScoreIterationListener` for the following:
```
o.d.o.l.ScoreIterationListener - Score at iteration 0 is NaN
o.d.o.l.ScoreIterationListener - Score at iteration 1 is NaN
```

This particular net model was working just fine in RC 3.8. Not sure if it's related to the fact that I'm now using an `EarlyStoppingListener`.

**Comment by agibsonccc** [comment]
Could you try this again with recent changes?

**Comment by crockpotveggies** [comment]
Still getting `NaN`. I fixed an exception at runtime that told me that `dropConnect()` was incorrectly set to `true` which I thought may have fixed the problem. However, it still persists. What debugging information can I provide?

**Comment by crockpotveggies** [comment]
@agibsonccc also noticed you tagged it as GUI but I'm seeing this in `ScoreIterationListener` as well

**Comment by crockpotveggies** [comment]
The problem causing https://github.com/deeplearning4j/nd4j/issues/897#issuecomment-217370843 also appears related to this problem as well.

**Comment by raver119** [comment]
Issue is pretty outdated, so closing it now.

**Comment by webstorms** [comment]
I've got the same issue that of Score at iteration X is NaN and cannot find any fix. How did you manage to overcome this issue?

**Comment by ChenJiaoyan** [comment]
I also got the same issue that Score at iteration X is NaN. I search with Google, but still have not found a solution.

**Comment by crockpotveggies** [comment]
For future users viewing this issue, documentation is going to be your best friend here: 

https://deeplearning4j.org/troubleshootingneuralnets
https://deeplearning4j.org/visualization

**Comment by sascha08-15** [comment]
Facing the same issue here (not UI related). I did have a look at all hints and related bugs. I have the same issue wit hthe ScoreIterationListener.

Class ScoreIterationListener methid iterationDone says: Score at iteration 0 is NaN
This holds true for subsequent iterations and does not go away (tried until 30000).

What I did so far is

- reviewed network config, reduced number of nodes in the net layers
- ensured input features / outputs are between -1 and 1: did apply normalizers, etc.
- exchange GPU with CPU backend
I read all (hope I really got all!) defects around NaN scores on a model. Am I missing something obvious here?

Network looks as follows:

final int lstmLayerSize = 10; // reduced this. was 250, 150 and 50
final int tbpttLength = 30; //also reduced this just to play around
return new NeuralNetConfiguration.Builder()
    .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
    .iterations(iterations)
    .learningRate(learningRate).seed(12345)
    .weightInit(WeightInit.XAVIER).updater(Updater.RMSPROP).rmsDecay(0.95).list()
    .layer(0,
        new GravesLSTM.Builder().nIn(numInputs).nOut(lstmLayerSize)
            .activation(Activation.SIGMOID).build())
    .layer(1,
        new GravesLSTM.Builder().nIn(lstmLayerSize).nOut(lstmLayerSize / 2)
            .activation(Activation.SIGMOID).build())
    .layer(2,
        new RnnOutputLayer.Builder(LossFunctions.LossFunction.MSE)
            .activation(Activation.SIGMOID)
            .nIn(lstmLayerSize / 2).nOut(numOutputs).build())
    .backpropType(BackpropType.TruncatedBPTT).tBPTTForwardLength(tbpttLength)
    .tBPTTBackwardLength(tbpttLength).pretrain(false).backprop(true)
    .build();

**Comment by raver119** [comment]
@sascha08-15 your problem isn't a bug or something. your net just doesn't learn, and math goes off somewhere internally. Check those 2 links above in this issue.

**Comment by raver119** [comment]
p.s. hey @crockpotveggies ! :)

**Comment by crockpotveggies** [comment]
Also suggest giving https://github.com/deeplearning4j/Arbiter a shot here since you have a few hyperparameters that could benefit from a search.
